pyanaconda/install_tasks.py

```diff
@@ -0,0 +1,400 @@
+
+    @property
+    def current_task_number(self):
+        with self._lock:
+            return self._current_task_number
+
+    @property
+    def current_queue_number(self):
+        with self._lock:
+            return self._current_queue_number
+
+    @property
+    def progress(self):
+        """Task queue processing progress.
+
+        The progress is reported as a floating point number from 0.0 to 1.0.
+        :returns: task queue processing progress
+        :rtype: float
+        """
+        with self._lock:
+            if self.current_task_number:
+                return self.task_count / self.current_task_number
+            else:
+                return 0.0
+
+    @property
+    def summary(self):
+        """Return a multi-line summary of the contents of the task queue.
+
+        :returns: summary of task queue contents
+        :rtype: str
+        """
+        if self.parent is None:
+            message = "Top-level task queue: %s\n" % self.name
+            # this is the top-level queue, so add some "global" stats
+            message += "Number of task queues: %d\n" % self.queue_count
+            message += "Number of tasks: %d\n" % self.task_count
+            message += "Task & task group listing:\n"
+        else:
+            message = "Task queue: %s\n" % self.name
+        for item in self:
+            for line in item.summary.splitlines():
+                message += " %s\n" % line
+        # remove trailing newlines from the top level message
+        if self.parent is None and message[-1] == "\n":
+            message = message[0: -1]
+        return message
+
+    @property
+    def parent(self):
+        return self._parent
+
+    @parent.setter
+    def parent(self, parent_item):
+        self._parent = parent_item
+        # Connect own signals "up" to the parent,
+        # so that it is possible to monitor how all nested TaskQueues and Tasks
+        # are running from the top-level element.
+
+        # connect own start/completion signal to parents queue start/completion signal
+        self.started.connect(self._parent.queue_started)
+        self.completed.connect(self._parent.queue_completed)
+
+        # propagate start/completion signals from nested queues/tasks
+        self.queue_started.connect(self._parent.queue_started)
+        self.queue_completed.connect(self._parent.queue_completed)
+        self.task_started.connect(self._parent.task_started)
+        self.task_completed.connect(self._parent.task_completed)
+
+    # make sure the parent is set correctly when adding, inserting or setting items
+    def append(self, item):
+        item.parent = self
+        super(TaskQueue, self).append(item)
+
+    def insert(self, index, item):
+        item.parent = self
+        super(TaskQueue, self).insert(index, item)
+
+    def __setitem__(self, index, item):
+        item.parent = self
+        super(TaskQueue, self).__setitem__(index, item)
+
+    def start(self):
```

Comment:
I see it as starting the execution/processing of the Task or TaskQueue that then runs (as indicated by the `running` property) to completion (as indicated by the `completed` property). Also TaskQueues don't care what they are iterating over - they just call `start()` on it, so renaming it just for the TaskQueue would be problematic.

So I think `start` is fine. :)